data/transactions/logic/assembler.go

```diff
@@ -1387,8 +1513,10 @@ func (ops *OpStream) assemble(text string) error {
 		if ops.Version == assemblerNoVersion {
 			ops.Version = AssemblerDefaultVersion
 		}
+		if ops.versionedPseudoOps == nil {
+			ops.versionedPseudoOps = prepareVersionedPseudoTable(ops.Version)
+		}
```

Comment:
@iten-alg Can the highlighted lines be moved into `newOpStream`?

* I reviewed the section because the construction in `prepareVersionedPseudoTable` looks like a one-time setup per `assemble` invocation.  `newOpStream`'s intent is to provide a factory method `assemble` initialization.
* I realize the highlighted version check predates the PR.  And I don't know if there's a rationale to retain the check here.  It seems like _no_ though I raise the question to double check.
  * If a rationale exists, then I withdraw my question.

Reply:
We don't know the version until we begin processing the OpStream, and this construction only does the work for the version being used, so it has to happen late.

Reply:
@jannotti @iten-alg Thanks for the responses - I'm good to resolve here.